lib/Backends/Interpreter/InterpreterNodes.cpp

```diff
@@ -876,6 +876,78 @@ void InterpreterFunction::fwdGatherInst(const glow::GatherInst *I) {
   }
 }
 
+template <typename ElemTy>
+void InterpreterFunction::fwdGatherRangesInstImpl(
+    const glow::GatherRangesInst *I) {
+  Tensor *dataT = getTensor(I->getData());
+  auto &dataTy = dataT->getType();
+  Tensor *rangesT = getTensor(I->getRanges());
+  auto &rangesTy = rangesT->getType();
+  Tensor *outT = getTensor(I->getOutput());
+  Tensor *lengthsT = getTensor(I->getLengths());
+
+  // Offset into the output tensor that keeps track of where to start
+  // copying data.
+  size_t outP = 0;
+
+  unsigned dataElementSize = dataTy.getElementSize();
+  unsigned numExamples = rangesTy.dims()[0];
+  unsigned exampleSize = rangesTy.dims()[1];
+
+  // Keep track of the total number of elements gathered across all
+  // examples for a sanity check later.
+  size_t grandTotalLen = 0;
+
+  // For each example in ranges:
+  for (size_t example = 0; example < numExamples; ++example) {
+    // Keep a running total of the lengths of all ranges in this example
+    // to record into lengthsT once the entire example is processed.
+    ElemTy totalLen = 0;
+
+    // For each range in the example:
+    for (size_t range = 0; range < exampleSize; ++range) {
+      // Get the start index and range length.
+      ElemTy start = rangesT->getHandle<ElemTy>().at({example, range, 0});
+      ElemTy len = rangesT->getHandle<ElemTy>().at({example, range, 1});
+
+      // Add the length of this current range to the example length counter.
+      totalLen += len;
+
+      // Copy the specified data to outT.
+      std::copy(&dataT->getUnsafePtr()[start * dataElementSize],
+                &dataT->getUnsafePtr()[(start + len) * dataElementSize],
+                &outT->getUnsafePtr()[outP]);
```

Comment:
Do we have some sort of check on whether this will go out of bound?

Reply:
The TensorHandle API has bounds checks, but raw pointer access does not.

Reply:
Yeah, so here it looks like raw pinter access. Maybe add some checks?